valhalla/sif/dynamiccost.h

```diff
@@ -601,26 +605,25 @@ class DynamicCost {
             // If not, we should keep looking
 
             // We are in range at the time we are allowed at this edge
-            if (access_type == baldr::AccessType::kTimedAllowed)
-              return true;
-            else if (access_type == baldr::AccessType::kDestinationAllowed)
-              return allow_conditional_destination_ || is_dest;
-            else
-              return false;
+            switch (access_type) {
+              case baldr::AccessType::kTimedAllowed:
+                restricted = false;
```

Comment:
this was a bug: we shouldn't `return true` within this loop. later restrictions might be `false` and that's what we want to get to. however, `return false` within the loop is fine of course